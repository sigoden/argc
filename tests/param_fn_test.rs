#[test]
fn test_param_fn_empty() {
    snapshot_spec!(&["spec", "_fn_args"]);
}

#[test]
fn test_param_fn_space() {
    snapshot_spec!(&["spec", "_fn_args", " "]);
}

#[test]
fn test_param_fn_args() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 abc"]);
}

#[test]
fn test_param_fn_args_dup_flag() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 -f"]);
}

#[test]
fn test_param_fn_args_dup_option() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 -o 5"]);
}

#[test]
fn test_param_fn_args_dup_dashdash1() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 -- abc"]);
}

#[test]
fn test_param_fn_args_dup_dashdash2() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 abc --"]);
}

#[test]
fn test_param_fn_args_dup_dashdash3() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 abc -- def"]);
}

#[test]
fn test_param_fn_args_incomplete_option() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o"]);
}

#[test]
fn test_param_fn_args_nontmatch_subcommand() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_prefer"]);
}

#[test]
fn test_param_fn_args_unknown_option() {
    snapshot_spec!(&["spec", "_fn_args", "cmd_preferred -f -o 4 -x"]);
}

#[test]
fn test_param_fn_bars() {
    snapshot_spec!(&["spec", "_fn_bars",]);
}
