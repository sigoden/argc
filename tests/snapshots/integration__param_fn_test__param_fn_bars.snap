---
source: tests/param_fn_test.rs
expression: output
---
RUN
spec _fn_bars

OUTPUT
argc__line=''
argc__words=(  )
_fn_bars;exit;

